### PR TITLE
fix top_retain TypeError exception

### LINKED ISSUE #183: top_retain results in TypeError: Cannot read property '_top' of undefined
**Bug report**

**Version (complete output of `terser -V`)**

3.10.12

**Complete CLI command or `minify()` options used**

... top_retain ...

**`terser` input**

**`terser` output or error**

```
TypeError: Cannot read property '_top' of undefined
    at Zn (/Users/standa/Prg/rollup-plugin-espruino-modules/node_modules/terser/dist/bundle.js:44:164713)
    at /Users/standa/Prg/rollup-plugin-espruino-modules/node_modules/terser/dist/bundle.js:44:184171
    at AST_Call.optimize (/Users/standa/Prg/rollup-plugin-espruino-modules/node_modules/terser/dist/bundle.js:44:121279)
    at Ft.before (/Users/standa/Prg/rollup-plugin-espruino-modules/node_modules/terser/dist/bundle.js:44:121036)
    at AST_Call.transform (/Users/standa/Prg/rollup-plugin-espruino-modules/node_modules/terser/dist/bundle.js:44:78398)
    at /Users/standa/Prg/rollup-plugin-espruino-modules/node_modules/terser/dist/bundle.js:44:78534
```

**Expected result**

### PR DESCRIPTION
**Pull request by kzc**
fixes #183

### CODE REVIEW

File: lib/compress.js [hunk 1]
```diff
@@ -4556,6 +4556,7 @@ merge(Compressor.prototype, {
 
     function retain_top_func(fn, compressor) {
         return compressor.top_retain
+            && fn
             && fn._top
```
Comment (opichals) [inline lib/compress.js]: nit: just doing the instanceof first would make the subsequent `fn.*` valid.

```suggestion
            && fn instanceof AST_Defun
            && fn._top
```

Comment (kzc) [reply lib/compress.js]: done

**Comment by kzc** [comment]
This PR is done.

**Comment by fabiosantoscode** [comment]
Looking great

**Pull request merged**